lua/barbar/state.lua

```diff
@@ -38,6 +46,7 @@ local WARN = severity.WARN
 
 --- @class barbar.state.buffer.data
 --- @field closing boolean whether the buffer is being closed
+--- @field will_close boolean whether the buffer will be closed
```

Comment:
This is the fix per-se for #550, we mark the buffer to avoid focusing it again.

Reply:
It would feel cleaner but I noticed some issues when re-using `closing` and I don't have enough time to make the change and debug possible issues, I'll leave as it is for now, we can refactor later.

Reply:
> I noticed some issues when re-using closing and I don't have enough time to make the change and debug possible issues, I'll leave as it is for now, we can refactor later

Sounds good. 

As a note for later, bit flags might be better (e.g. `bit.band(is_open, state.closing)`)